docs/design/features/typemap.md

```diff
@@ -0,0 +1,223 @@
+# Interop Type Map
+
+## Background
+
+When interop between languages/platforms involves the projection of types, some kind of type mapping logic must often exist. This mapping mechanism is used to determine what .NET type should be used to project a type from language X and vice versa.
+
+The most common mechanism for this is the generation of a large look-up table at build time, which is then injected into the application or Assembly. If injected into the Assembly, there is typically some registration mechanism for the mapping data. Additional modifications and optimizations can be applied based on the user experience or scenarios constraints (that is, build time, execution environment limitations, etc).
+
+Prior to .NET 10 there were at least three (3) bespoke mechanisms for this in the .NET ecosystem:
+
+* C#/WinRT - [Built-in mappings](https://github.com/microsoft/CsWinRT/blob/master/src/WinRT.Runtime/Projections.CustomTypeMappings.tt), [Generation of vtables for AOT](https://github.com/microsoft/CsWinRT/blob/b1733e95c6d35b551fc8cf6fe04e2a0c287346dd/src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs#L1597).
```

Comment:
> (https://github.com/microsoft/CsWinRT/blob/master/src/WinRT.Runtime/Projections.CustomTypeMappings.tt)

This is a non-commit based link. It will not age well if the file is renamed, removed or moved.

Reply:
Same for some below.